src/rgw/rgw_rados.cc

```diff
@@ -8137,6 +8137,13 @@ int RGWRados::Object::Delete::delete_obj()
 
   if (params.versioning_status & BUCKET_VERSIONED || explicit_marker_version) {
     if (instance.empty() || explicit_marker_version) {
+      if (!instance.empty() && explicit_marker_version) {
```

Comment:
@dongbula (NIT) Please consider keeping PR title < 50 characters.
https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits